python/pyarrow/parquet.py

```diff
@@ -871,21 +889,34 @@ def filter_accepts_partition(self, part_key, filter, level):
         if p_column != f_column:
             return True
 
-        f_type = type(f_value)
+        p_value = self.levels[level].dictionary[p_value_index].as_py()
 
-        if isinstance(f_value, set):
+        if op in {'in', 'not in'}:
+            if not (isinstance(f_value, Container) and
+                    isinstance(f_value, Iterable)):
+                raise ValueError("Op '%s' requires an iterable such as list,"
+                                 " set or tuple", op)
             if not f_value:
-                raise ValueError("Cannot use empty set as filter value")
-            if op not in {'in', 'not in'}:
-                raise ValueError("Op '%s' not supported with set value",
-                                 op)
+                raise ValueError("Cannot use empty iterable as filter value")
+
             if len({type(item) for item in f_value}) != 1:
-                raise ValueError("All elements of set '%s' must be of"
+                raise ValueError("All elements of the iterable '%s' must be of"
                                  " same type", f_value)
             f_type = type(next(iter(f_value)))
 
-        p_value = f_type(self.levels[level]
-                         .dictionary[p_value_index].as_py())
+            p_value = f_type(p_value)
+
+            if op == 'in':
+                return p_value in f_value
+            else:
+                return p_value not in f_value
```

Comment:
Could you only do the `f_value` checking and `p_value` preparation in this `if` branch, but keep the actual check below?